File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.go
func twoSum(numbers []int, target int) []int {
    leftPointer := 0
	rightPointer := len(numbers) - 1

	for {
		temp := numbers[leftPointer] + numbers[rightPointer]

		if temp == target {
			return []int{leftPointer + 1, rightPointer + 1}
		}

		if temp > target {
			rightPointer -= 1
		} else {
			leftPointer += 1
		}
	}
}

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.rs
impl Solution {
    pub fn two_sum(numbers: Vec<i32>, target: i32) -> Vec<i32> {
        let mut left_pointer = 0;
        let mut right_pointer = numbers.len() - 1;

        loop {
            let temp = numbers[left_pointer] + numbers[right_pointer];

            if temp == target {
                return vec![(left_pointer + 1) as i32, (right_pointer + 1) as i32];
            }

            if temp > target {
                right_pointer -= 1;
            } else {
                left_pointer += 1;
            }
        }
    }
}